apps/music/test/marionette/playlist_test.js

```diff
@@ -276,5 +276,86 @@ marionette('Music player playlist', function() {
       assert.equal(PlaylistHelper.songTitle(songs[0]),
                    'The Ecuadorian Embassy');
     });
+
+    test('Most played playlist sort order', function() {
+      music.launch();
+      music.waitForFirstTile();
+
+      music.switchToAlbumsView();
+
+      music.selectAlbum('We crash computers');
+
+      music.waitForSongs(function(songs) {
+        return songs.length >= 6;
+      });
+
+      var songs = music.songs;
+      assert.equal(songs.length, 6);
+
+      // we set the playcount.
+      client.executeScript(function() {
+        var w = window.wrappedJSObject;
+        var songData = w.SubListView.dataSource[4];
+        songData.metadata.played = 2;
+        w.musicdb.updateMetadata(songData.name, songData.metadata);
+
+        // 'Crash'
+        songData = w.SubListView.dataSource[3];
+        songData.metadata.played = 4;
+        w.musicdb.updateMetadata(songData.name, songData.metadata);
+
+        songData = w.SubListView.dataSource[2];
+        songData.metadata.played = 3;
+        w.musicdb.updateMetadata(songData.name, songData.metadata);
+
+        // 'XOXO'
+        songData = w.SubListView.dataSource[1];
+        songData.metadata.played = 5;
+        w.musicdb.updateMetadata(songData.name, songData.metadata);
+
+        // 'Yield to Thread'
+        songData = w.SubListView.dataSource[0];
+        songData.metadata.played = 1;
+        w.musicdb.updateMetadata(songData.name, songData.metadata);
+
+        // 'Abort'
+        // Play count is 0 for that song.
```

Comment:
Nit: the code in this executeScript call might be clearer if you re-ordered it to go from highest to lowest playcount.